### PR TITLE
KAFKA-12796: Removal of deprecated classes under `streams-scala`

### PR DESCRIPTION
**Pull request by jlprat**
Removes previously deprecated methods in older KIPs

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KTable.scala [hunk 1]
```diff
@@ -304,7 +304,7 @@ class KTable[K, V](val inner: KTableJ[K, V]) {
     new KStream(inner.toStream[KR](mapper.asKeyValueMapper, named))
 
   /**
-   * Suppress some updates from this changelog stream, determined by the supplied [[Suppressed]] configuration.
+   * Suppress some updates from this changelog stream, determined by the supplied [[org.apache.kafka.streams.kstream.Suppressed]] configuration.
```
Comment (jlprat) [inline streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KTable.scala:307]: Use the right non-deprecated class

**Comment by jlprat** [comment]
cc @cadonna PR is ready for review, if you would like to take a look. Thanks!

**Comment by jlprat** [comment]
As discussed with @ableegoldman only classes deprecated since at least 2.5 can be removed. For this reason, `Serdes` can't be removed yet.

**Review by cadonna — APPROVED** [review]
Thank you for the PR! @jlprat 

LGTM!

**Comment by cadonna** [comment]
Failed test is unrelated and known to be flaky:
```
kafka.server.RaftClusterTest.testCreateClusterAndCreateListDeleteTopic()
```

**Comment by jlprat** [comment]
Thanks for the review @cadonna !

**Pull request merged**